### ISSUE TITLE
Zigbee AD push buttons not generating devices ?

### ISSUE THREAD

**Issue by PVerburg** [domoticz/domoticz#6105]
Buttons not auto configured.

Not sure if I supplied the right information I have 3 different zigbee buttons but all have the same problem ?

homeassistant/device_automation/0xa4c138b11aefa462/action_4_single/config

{
  "automation_type": "trigger",
  "device": {
    "identifiers": [
      "zigbee2mqtt_0xa4c138b11aefa462"
    ],
    "manufacturer": "TuYa",
    "model": "Wireless switch with 4 buttons (TS0044)",
    "name": "moment-4",
    "sw_version": "0122052017",
    "via_device": "zigbee2mqtt_bridge_0xe0798dfffeb321b3"
  },
  "origin": {
    "name": "Zigbee2MQTT",
    "sw": "1.38.0",
    "url": "https://www.zigbee2mqtt.io"
  },
  "payload": "4_single",
  "subtype": "4_single",
  "topic": "zigbee2mqtt/moment-4/action",
  "type": "action"
}

device_automation
0xa4c138b11aefa462
action_1_single
config = {"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"1_single","subtype":"1_single","topic":"zigbee2mqt…
action_2_single
config = {"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"2_single","subtype":"2_single","topic":"zigbee2mqt…
action_3_single
config = {"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"3_single","subtype":"3_single","topic":"zigbee2mqt…
action_4_single
config = {"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"4_single","subtype":"4_single","topic":"zigbee2mqt…
action_1_double
config = {"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"1_double","subtype":"1_double","topic":"zigbee2mqt…

Paul V.

**Comment by gizmocuz** [comment]
@PVerburg , thank you for the config object.
Is this the only config for this switch?

Could you use 'MQTT Explorer' and in the search box type '0xa4c138b11aefa462'

I would expect a switch config as well

In Domoticz in settings, you enabled the 'Accept new hardware Devices' ?
If you press a button, and go in Domoticz to the 'Devices' overview, you do not see them being added?

When you press a button, could you also send me the payload of

**zigbee2mqtt/moment-4/action**

**Comment by PVerburg** [comment]
Yes do do follow the procedure for adding devices been using Z2M for a year now but first time for the push buttons and auto config all the other-ones add perfectly i got temp door sensors and many power-plugs/routes/usage monitors .  in the past did manual mqtt addons but now have over 200 devices and becomes a too big a job. 

Thanks for your help.

Sorry for the screen capture but could not see another way of doing it.
![image](https://github.com/domoticz/domoticz/assets/7756751/367ac0c6-90cb-4695-abde-4b01707e95eb)


So two buttons and not behaving the same in there main topic but the action topic behavior is identical.

zigbee2mqtt 1.38.0

TuYa TS0044
topic
zigbee2mqtt/moment-4
value
{"action":"1_single","battery":100,"last_seen":"2024-06-28T12:44:25+12:00","linkquality":104,"voltage":3000}
immediately followed with
value
{"battery":100,"last_seen":"2024-06-28T12:44:25+12:00","linkquality":100,"voltage":3000}

topic
zigbee2mqtt/moment-4/action
value
1_single

up to 4_single
or 1_double or 1_hold up to 4_hold
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Linkoze  LKWSZ211
topic 
zigbee2mqtt/Button-1
value
{"action":"button_1_single","battery":100,"last_seen":"2024-06-28T12:39:00+12:00","linkquality":108}

topic
zigbee2mqtt/Button-1/action
value
button_1_single

**Comment by gizmocuz** [comment]
Thank you very much for the above information, but could you please copy/paste the config topics as well. I can't work with screenshots with partly json information

If you use MQTT Explorer, and select the config object from the left, on the top-right side of the application you see some buttons, one of them is 'copy' topic, a bit lower you will see the payload of the config object, also here is a 'copy' button

![image](https://github.com/domoticz/domoticz/assets/2265299/fc2e8074-fbfc-400c-8e26-af105b025315)

**Comment by PVerburg** [comment]
Sorry they where in my fist post but in complete.
topic
homeassistant/device_automation/0xa4c138b11aefa462/action_1_single/config
value
`{"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138b11aefa462"],"manufacturer":"TuYa","model":"Wireless switch with 4 buttons (TS0044)","name":"moment-4","sw_version":"0122052017","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"1_single","subtype":"1_single","topic":"zigbee2mqtt/moment-4/action","type":"action"}`

++++++++++++++++++++++++++++++++++++++++++++++++++++++
second device

topic
homeassistant/device_automation/0xa4c138bbc8972826/action_button_2_single/config

value
`{"automation_type":"trigger","device":{"identifiers":["zigbee2mqtt_0xa4c138bbc8972826"],"manufacturer":"Linkoze","model":"Wireless switch (2-key) (LKWSZ211)","name":"Button-1","via_device":"zigbee2mqtt_bridge_0xe0798dfffeb321b3"},"origin":{"name":"Zigbee2MQTT","sw":"1.38.0","url":"https://www.zigbee2mqtt.io"},"payload":"button_2_single","subtype":"button_2_single","topic":"zigbee2mqtt/Button-1/action","type":"action"}
`
hope this help.

**Comment by gizmocuz** [comment]
Thank you for the config object.

Did you enter '0xa4c138b11aefa462' (without the quotes) in the MQTT Explorers search box?
Normally there is also a light/binary/sensor config object and not only a device_automation.

Is there anywhere in zigbee2mqtt that you can configure for this node to send the normal config object?
You paired these buttons directly with zigbee2mqtt and not via another node?

**Comment by PVerburg** [comment]
After looking at this link  https://www.zigbee2mqtt.io/devices/LKWSZ211.html
It only expose "action"  not switch ?? must be a new "standard" I double checked in MQTT Explore and there are no other entries for these type of switches I have 3 different models/brands.
I'm not a programmer but maybe the relevant code block that handles switches could be copied and adjusted for the new "action" standard ?? if  you can point me in the right direction I could try to debug this for you if that helps?

**Comment by gizmocuz** [comment]
I assume you are on the latest zigbee2mqtt? (And I hope via docker-compose)

Well, for example, I have this one:

https://www.zigbee2mqtt.io/devices/TS0041.html#tuya-ts0041

Working perfectly

Internally it is a matter of allowing the 'device_automation' component but it should not be necessary.

Seems like the same issue from

https://github.com/domoticz/domoticz/issues/5668

**Comment by PVerburg** [comment]
Ok thanks for that link it dint help me solve the situation. I did load HA and connected my mqttserver to it. it did discover the buttons/switch and I could assign functions to them but I don't want to use HA. So I might have to resorted to a node-red solution until this feature is sorted. 
Thanks for your help.

**Comment by gizmocuz** [comment]
Maybe update your zigbee2mqtt?

You don't need a Node-Red solution. Maybe investigate what if different between the above Tuya and your switch (zigbee2mqtt wise, like the config object, and ......... OK ... scrap all this comment!

I have the exact same switch as you do as well! Tuya TS0044 ! And this is working perfectly.

![image](https://github.com/domoticz/domoticz/assets/2265299/4936d7aa-201f-4ea7-a18a-d12bc728f16a)

Default configuration in zigbee2mqtt, nothing strange configured, all default, never touched a thing for this node/switch in zigbee2mqtt.

I asked above if you have installed zigbee2mqtt via 'docker compose' (as that's the way) and you are using the latest version?
(I am using image: koenkk/zigbee2mqtt:latest-dev)

I just updated my zigbee2mqtt to version:

Zigbee2MQTT version: 1.39.0-dev commit: (https://github.com/Koenkk/zigbee2mqtt/commit/f97ad2e)
Coordinator type: zStack3x0
Coordinator revision: 20201026
Frontend version: 0.7.2, 19.64.0, 0.51.0

Checked the switch, working excellent.

As you can see below it has both a 'sensor' and a 'device_automation' component, where only the 'sensor' part is used in Domoticz.

![image](https://github.com/domoticz/domoticz/assets/2265299/5c7f8499-ba16-404f-ac4c-ab89bb866cc1)

So to recap, what version of Zigbee2mqtt are you using? Are you using the latest version? (And of course one of the latest Domoticz beta versions)

**Comment by PVerburg** [comment]
Thank you for your persistence I noticed you had an action under sensors I did not so it had to do with zigbee2mqtt and not Domoticz.  I did set up a new Pi4 with IOTstack i.e. Docker etc. a clean version of Zigbee2MQTT ver 1.39.0 and did some more testing. The answer was simple in the end as it always is, there is a flag called "Home Assistant legacy triggers"  if you enable this the action appears in the senor tree and now Domoticz will generate devices for the buttons.
When I was building this version I am using the Ember stack with latest drivers etc. and didn't want any legacy stuff that might be discontinued as the name suggests and forgot all about it as everything else worked as expected for the last 2 months or so until I purchased some of these buttons for testing to pick the "best" ones to implement to find none of them worked. 

Having established that Domoticz AD relies on a legacy option I suspect that in sometime to come the option might become unavailable so my request would be to update to the new action tree, or seek clarification why the change was made in HA and why the flag was put into Zigbee2MQTT ? 
I did not have that flag set when testing with HA and the devices showed up there i.e. they where discovered with the "new" standard.
Just checked to confirm the "new" automation status with the button info remains even when legacy is turned on so it basically doubles up, so if AD updates to the "new" system no changes are required on the zigbee2mqtt side, but obviously a re-detection has to be done to find all the "new" switch information in the new tree.

Again thanks for your time and I hope this information helps someone else to understand what and why the problem occurred I'm at a loss why it only affects wireless buttons but I'm sure HA had a good reason for it.

**Comment by gizmocuz** [comment]
Domoticz already supports the device_automation component but it is disabled internally for now.
These flags are enabled by default, so until this is not the case, please keep everything 'default' as it should be

Also in our wiki they are enabled in the screenshot

https://www.domoticz.com/wiki/Zigbee2MQTT
